src/Cassandra.Tests/Requests/StartupOptionsFactoryTests.cs

```diff
@@ -0,0 +1,64 @@
+﻿//
+//       Copyright (C) 2019 DataStax Inc.
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+//
+
+using System;
+using System.Reflection;
+
+using Cassandra.Helpers;
+using Cassandra.Requests;
+
+using NUnit.Framework;
+
+namespace Cassandra.Tests.Requests
+{
+    [TestFixture]
+    public class StartupOptionsFactoryTests
+    {
+        [Test]
+        public void Should_ReturnCorrectProtocolStartupOptions_When_OptionsAreSet()
+        {
+            var factory = new StartupOptionsFactory();
+
+            var options = factory.CreateStartupOptions(new ProtocolOptions().SetNoCompact(true).SetCompression(CompressionType.Snappy));
+
+            Assert.AreEqual(5, options.Count);
+            Assert.AreEqual("snappy", options["COMPRESSION"]);
+            Assert.AreEqual("true", options["NO_COMPACT"]);
+            Assert.AreEqual("DataStax C# Driver", options["DRIVER_NAME"]);
+            Assert.AreEqual("3.0.0", options["CQL_VERSION"]);
+            var assemblyVersion = MultiTargetHelpers.GetAssembly(typeof(Cluster)).GetCustomAttribute<AssemblyInformationalVersionAttribute>().InformationalVersion;
+            Assert.AreEqual(assemblyVersion, options["DRIVER_VERSION"]);
+            var indexOfVersionSuffix = assemblyVersion.IndexOf('-');
+            var versionPrefix = indexOfVersionSuffix == -1 ? assemblyVersion : assemblyVersion.Substring(0, indexOfVersionSuffix);
+            var version = Version.Parse(versionPrefix);
+            Assert.LessOrEqual(version.Major, 4);
```

Comment:
NIT: I think comparing against assembly version (and validating that assembly version contains x.y.z >= 1.0 is enough. I personally prefer having a looser test than having to adapt it on both drivers by version.

In any case, comparisons between `Version` instances can be used: `Assert.Greater(version1, new Version(1, 0))`.

Reply:
Makes sense, I've changed it so that it doesn't have to change in dse driver and uses `Version` instances for comparison.